tests/integration/tests/test_config_propagation.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+from typing import List
+
+import pytest
+from test_util import harness, util
+
+LOG = logging.getLogger(__name__)
+
+
+@pytest.mark.node_count(3)
+def test_config_propagation(instances: List[harness.Instance]):
```

Comment:
note that this test will also work without our controller (since nodes get the right dns config when they join).

make sure to change the config then validate the new config is propagated (e.g. edit the cluster-domain)